iree/hal/cts/command_buffer_test.cc

```diff
@@ -0,0 +1,284 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstring>
+#include <vector>
+
+#include "iree/base/status.h"
+#include "iree/hal/cts/cts_test_base.h"
+#include "iree/hal/testing/driver_registry.h"
+#include "iree/testing/gtest.h"
+#include "iree/testing/status_matchers.h"
+
+namespace iree {
+namespace hal {
+namespace cts {
+
+using ::testing::ContainerEq;
+
+class CommandBufferTest : public CtsTestBase {
+ protected:
+  static constexpr iree_device_size_t kBufferSize = 16;
+
+  // Helper for submitting a command buffer and waiting for it to complete.
+  // |signal_semaphore| must start with a payload value of 0.
+  void SubmitAndWait(iree_hal_command_buffer_t* command_buffer,
```

Comment:
You want to return the status from this instead of asserting inside of it - ASSERT just `return;`s from the current function, which means if this fails then the caller will continue executing (vs exiting, which is what you want). EXPECTS are fine in non-test functions, but the asserts aren't (unless you then *also* have a return to bail, but at that point you aren't really getting much from having both).

Reply:
Argh, I thought `ASSERT` macros would actually `assert()` ... reworked to return status objects.